src/main/java/ch/njol/skript/expressions/ExprTool.java

```diff
@@ -1,73 +1,77 @@
 @Description("The item an entity is holding in their main or off hand.")
-@Examples({"player's tool is a pickaxe",
-	"player's off hand tool is a shield",
-	"set tool of all players to a diamond sword",
-	"set offhand tool of target entity to a bow"})
+@Example("""
+	player's tool is a pickaxe
+	player's off hand tool is a shield
+	set tool of all players to a diamond sword
+	set offhand tool of target entity to a bow
+""")
 @Since("1.0")
 public class ExprTool extends PropertyExpression<LivingEntity, Slot> {
+
 	static {
 		Skript.registerExpression(ExprTool.class, Slot.class, ExpressionType.PROPERTY,
-			"[the] ((tool|held item|weapon)|1¦(off[ ]hand (tool|item))) [of %livingentities%]",
-			"%livingentities%'[s] ((tool|held item|weapon)|1¦(off[ ]hand (tool|item)))");
+			"[the] (tool|held item|weapon) [of %livingentities%]",
+			"%livingentities%'[s] (tool|held item|weapon)",
+			"[the] off[ ]hand (tool|item) [of %livingentities%]",
+			"%livingentities%'[s] off[ ]hand (tool|item)"
+		);
 	}
 
 	private boolean offHand;
 
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+		//noinspection unchecked
 		setExpr((Expression<LivingEntity>) exprs[0]);
-		offHand = parser.mark == 1;
+		offHand = matchedPattern >= 2;
 		return true;
 	}
 
 	@Override
 	protected Slot[] get(Event event, LivingEntity[] source) {
-		final boolean delayed = Delay.isDelayed(event);
+		boolean delayed = Delay.isDelayed(event);
 		return get(source, entity -> {
 			if (!delayed) {
 				if (!offHand && event instanceof PlayerItemHeldEvent playerItemHeldEvent
 					&& playerItemHeldEvent.getPlayer() == entity) {
-					final PlayerInventory i = playerItemHeldEvent.getPlayer().getInventory();
-					return new InventorySlot(i, getTime() >= 0 ? playerItemHeldEvent.getNewSlot() : playerItemHeldEvent.getPreviousSlot());
+
+					PlayerInventory inventory = playerItemHeldEvent.getPlayer().getInventory();
+					return new InventorySlot(inventory, getTime() >= 0 ? playerItemHeldEvent.getNewSlot() : playerItemHeldEvent.getPreviousSlot());
+
 				} else if (event instanceof PlayerBucketEvent playerBucketEvent
 					&& playerBucketEvent.getPlayer() == entity) {
-					final PlayerInventory i = playerBucketEvent.getPlayer().getInventory();
-					boolean isOffHand = playerBucketEvent.getHand() == org.bukkit.inventory.EquipmentSlot.OFF_HAND || offHand;
-					return new InventorySlot(i, isOffHand ? EquipmentSlot.EquipSlot.OFF_HAND.slotNumber
-						: playerBucketEvent.getPlayer().getInventory().getHeldItemSlot()) {
+
+					PlayerInventory inventory = playerBucketEvent.getPlayer().getInventory();
+					boolean isOffHand = offHand || playerBucketEvent.getHand() == org.bukkit.inventory.EquipmentSlot.OFF_HAND;
+					int inventorySlot = isOffHand ? EquipmentSlot.EquipSlot.OFF_HAND.slotNumber : inventory.getHeldItemSlot();
+
+					return new InventorySlot(inventory, inventorySlot) {
```

Comment:
This part of the code is a bit weird to me, is anyone able to provide me a reasoning for this? It's forced to use the old EquipSlot enum currently which I would like to remove but without any clear case of reasoning I find it confusing.

This little snippet of code also breaks the expression currently if someone does `tool of player` but `getHand()` in the event returns offhand.

Reply:
i mean i'm not sure why you say it needs to use the old enum? It doesn't seem like there's anything that forces its use here.

I expect this was intended so that when you do
```
on bucket fill:
    send tool of player
```
it gets the tool used in the event, the bucket, but I think that's a poor solution and a better option would be to have a way to get which hand was used.

Reply:
> i mean i'm not sure why you say it needs to use the old enum? It doesn't seem like there's anything that forces its use here.

Saying forced is relatively wrong on my end, but mainly design wise the reasoning for it using it was for `.slotNumber` as it used InventorySlot, if we're fine with using the slot number literals that would be fine by me.

Reply:
You could just expose that map from EquipmentSlot / move it to a full utils method

Reply:
I've gone ahead and moved it into BukkitUtils as a utils method, depending on whether this or #7713 gets merged first, the other will run into conflicts.